### PR TITLE
Update ConfigDef to have more metadata and use it for validation

### PR DESCRIPTION
**Pull request by varunpuranik**
- Added more metadata to the connector ConfigDef for use with control center
- Extending from AbstractConfig to be able to use it for parsing and validation.
- Also marked Kafka dependencies as "provided" in sbt to avoid packaging them in the jar.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -19,8 +19,8 @@ libraryDependencies ++= {
   val json4sVersion = "3.5.0"
 
   Seq(
-    "org.apache.kafka" % "connect-api" % kafkaVersion,
-    "org.apache.kafka" % "connect-json" % kafkaVersion,
+    "org.apache.kafka" % "connect-api" % kafkaVersion % "provided",
+    "org.apache.kafka" % "connect-json" % kafkaVersion % "provided",
     "com.microsoft.azure" % "azure-eventhubs" % azureEventHubSDKVersion,
```
Comment (dluc) [inline build.sbt:24]: does the provided package match the version required? I'm wondering what happens if we expect version "foo.X" and the provided is "foo.Y"

File: src/test/scala/com/microsoft/azure/iot/iothub2kafka/IotHubSourceTaskTest.scala [hunk 2]
```diff
@@ -11,6 +11,8 @@ import org.scalatest.{FlatSpec, GivenWhenThen}
 
 class IotHubSourceTaskTest extends FlatSpec with GivenWhenThen with JsonSerialization {
 
+  // This test is ignored as the configuration values in the test application.conf has no values.
```
Comment (dluc) [inline src/test/scala/com/microsoft/azure/iot/iothub2kafka/IotHubSourceTaskTest.scala]: nit: typo s/has/have/

**Review by dluc — APPROVED** [review]
Just a question about the behavior of "provided" dependencies

File: build.sbt (hunk 1 above)
Comment (varunpuranik) [reply build.sbt:24]: The connector expects Kafka v0.10 to be installed. I have added a line to the README to say so.

**Pull request merged**